tools/cldr-code/src/main/java/org/unicode/cldr/tool/PathInfo.java

```diff
@@ -0,0 +1,149 @@
+            if (path.startsWith("//")) {
+                showPath(path, file);
+            } else {
+                if (!prePopulated && file != null) {
+                    // scan all possible XPaths
+                    System.err.println("INFO: Scanning all possible xpaths for hex IDs...");
+                    // TODO: include fullpath, etc etc...
+                    file.fullIterable().forEach(x -> StringId.getHexId(x));
+                    prePopulated = true;
+                }
+                final String xpath = StringId.getStringFromHexId(path);
+                if (xpath == null) {
+                    System.err.println("• ERROR: could not find hex id " + path + " - may not have been seen yet.");
+                    hadStringMisses = true;
+                }
+                showPath(xpath, file);
+            }
+        }
+        if (hadStringMisses) {
+            System.err.println("ERROR: One or more XPaths could not be converted from hex form.");
+            if(file == null) {
+                System.err.println("Tip: Set a locale ID with -l when using hex ids");
+            }
+            System.exit(1);
+        }
+    }
+
+    private static void showPath(final String path, CLDRFile file) {
+        System.out.println("-------------------\n");
+        System.out.println("• " + path);
+        System.out.println("• " + StringId.getHexId(path));
+        // TODO: PathHeader, etc.
+        if (file == null) {
+            return;
+        }
+        final String dPath = CLDRFile.getDistinguishingXPath(path, null);
+        if (dPath != null && !dPath.equals(path)) {
+            System.out.println("• Distinguishing: " + dPath);
+            System.out.println("• Distinguishing:" + StringId.getHexId(dPath));
+        }
+        System.out.println("• Value: " + file.getStringValue(dPath));
+        // File lookup
+        SourceLocation source = file.getSourceLocation(path);
+        if (source != null) {
+            System.out.println(source + " XML Source Location");
+        }
+        // Inheritance lookup
+        final String fullPath = file.getFullXPath(path);
+        if (!fullPath.equals(path)) {
+            System.out.println("• Full path: " + fullPath);
+            System.out.println("• Full path: " + StringId.getHexId(fullPath));
+        }
+
+        Status status = new Status();
+        if (false) {
+            // For debugging: compare the below to calling getSourceLocaleIdExtended directly.
+            final String xlocale = file.getSourceLocaleIdExtended(dPath, status, true, null);
+            System.out.println("• SLIE = " + xlocale+":"+status.pathWhereFound);
+        }
+        System.out.println("• Inheritance chain:");
+        for (final LocaleInheritanceInfo e : file.getPathsWhereFound(dPath)) {
+            System.out.println("\n  • " + e); // reason : locale + xpath
+            if (e.getLocale() != null && e.getPath() != null && !e.getLocale().equals("code-fallback")) {
```

Comment:
"code-fallback" must have a constant somewhere...

CLDRResolver.java:
```
    public static final String CODE_FALLBACK = "code-fallback"
```

also XMLSource.java:
```
    public static final String CODE_FALLBACK_ID = "code-fallback";
```

Reply:
somewheres… ok, will use that.

I think for this API it would be better to actually omit the (pseudo) locale, because `codefallback` is already the reason code.

Reply:
OK, I still need this test, because the constructed path has to have a locale.

Reply:
BTW, it is too hard to find constants like this, which makes it less likely they'll be used. Suggest we have a single folder for files that just have constants in them (eg no dependencies on any code).

Reply:
or even a single file, and then static import them. We have LDMLConstants.java in the icu package, we could move and repurpose it.  and move all constants there.

Reply:
a better name would just be Constants.java